src/components/CollectionsAndPalettes/common/ItemCard/ItemCard.tsx

```diff
@@ -1,43 +1,111 @@
 import * as React from "react";
 import cn from "classnames";
 import styles from "../../common/ItemCard/ItemCard.module.scss";
+import { CollectionItemT } from "../../../../constants/types";
+import { AppContext, AppContextState } from "../../../../context/AppContext";
+import { get } from "lodash-es";
 
 interface Props {
   mode: any;
-  item: {
-    type: any;
-    title1: string;
-    title2: string;
-    title3: string;
-    price: string;
-    imagePath: string;
-  };
+  item: CollectionItemT;
 }
 
 export default class ItemCard extends React.Component<Props, any> {
-  renderImageItem(type) {
+  static contextType = AppContext;
+  context!: AppContextState;
+  materialItem: any;
+
+  mapAlgoleaToObject(): any {
+    let item = null;
+    if (this.props.item.material) {
+      item = this.props.item.material.id.toString();
+    } else {
+      item = this.props.item.upload;
+    }
+    const algoliaProduct = this.context.productsCache.getProduct(item);
+    const color = get(
+      algoliaProduct,
+      "data.color",
+      "this.props.item.material.color"
+    );
+    const manufacturer = get(
+      algoliaProduct,
+      "data.manufacturer",
+      "this.props.item.material.manufacturer"
+    );
+    const imageUrl = get(
+      algoliaProduct,
+      "data.thumbnail_url",
+      "https://dev.design.shop/static/version1613493863/frontend/Magento/luma/en_US/Magento_Catalog/images/product/placeholder/image.jpg"
+    );
+    const priceSign = get(
+      algoliaProduct,
+      "data.price_sign",
+      "this.props.item.material.price_sign"
+    );
+    const name = get(
+      algoliaProduct,
+      "data.name",
+      "this.props.item.material.name"
+    );
+    this.materialItem = {
```

Comment:
Storing this item on the class could, in some cases, make it so that we don't display the fields properly or we display out-of-date ones. (It is quite unlikely in this scenario, but I still think we should just call this method `getMaterial` or something like that, make it return this object instead of storing it on the class and just call it in the renderMaterialInfo() and renderMaterialImage() functions to make sure that we always have the latest data.